mlx-fast/src/lib.rs

```diff
@@ -0,0 +1,267 @@
+use mlx_internal_macros::default_device;
+use mlx_rs::{Array, Stream, StreamOrDevice};
+
+// TODO: Implement custom kernels:
+// https://github.com/ml-explore/mlx-swift/blob/d649c62b77c487c25012910b0d02b30283d388ca/Source/MLXFast/MLXFastKernel.swift
+
+/// Optimized implementation of `NN.RoPE`.
+#[allow(non_snake_case)]
+#[default_device]
+pub fn RoPE_device<'a>(
+    array: impl AsRef<Array>,
+    dimensions: i32,
+    traditional: bool,
+    base: impl Into<Option<f32>>,
+    scale: f32,
+    offset: i32,
+    freqs: impl Into<Option<&'a Array>>,
+    stream: impl AsRef<Stream>,
+) -> Array {
+    let base = base.into();
+    let base = mlx_sys::mlx_optional_float {
+        value: base.unwrap_or(0.0),
+        has_value: base.is_some(),
+    };
+
+    unsafe {
+        let new_array = mlx_sys::mlx_fast_rope(
```

Comment:
I haven't looked into the c++ side. Just curious, do we need to catch errors here?